### ISSUE TITLE
Need some help to count the number of fruits in the garden

### ISSUE THREAD

**Issue by leminhbk** [cvlab-stonybrook/LearningToCountEverything#4]
Dear Doctor,

I used this model to count the number of fruit in the garden but the result is not good.
Please help me to improve the model in order to make it more accuracy.

**The input images**
![01s](https://user-images.githubusercontent.com/28331455/121831479-bfae7c00-ccf1-11eb-87bb-4b724bbecdfa.jpg)
![02](https://user-images.githubusercontent.com/28331455/121831480-c1783f80-ccf1-11eb-9fe7-a01ecbdf8395.jpg)
![03](https://user-images.githubusercontent.com/28331455/121831481-c210d600-ccf1-11eb-8431-abe9840df847.jpg)
![04](https://user-images.githubusercontent.com/28331455/121831484-c210d600-ccf1-11eb-99ed-d7b9bc67f4df.jpg)
![05](https://user-images.githubusercontent.com/28331455/121831485-c2a96c80-ccf1-11eb-83d3-ffeed3abdad6.jpg)
![06](https://user-images.githubusercontent.com/28331455/121831488-c3420300-ccf1-11eb-88d0-d7609e5e01e4.jpg)

**The output images**
[01s_box.txt](https://github.com/cvlab-stonybrook/LearningToCountEverything/files/6645198/01s_box.txt)
![01s_out](https://user-images.githubusercontent.com/28331455/121831219-267f6580-ccf1-11eb-9079-e990127e2f4c.png)
[02_box.txt](https://github.com/cvlab-stonybrook/LearningToCountEverything/files/6645199/02_box.txt)
![02_out](https://user-images.githubusercontent.com/28331455/121831226-297a5600-ccf1-11eb-96bc-4bae22a0160e.png)
[03_box.txt](https://github.com/cvlab-stonybrook/LearningToCountEverything/files/6645200/03_box.txt)
![03_out](https://user-images.githubusercontent.com/28331455/121831228-2a12ec80-ccf1-11eb-971b-6e911a574dc9.png)
[04_box.txt](https://github.com/cvlab-stonybrook/LearningToCountEverything/files/6645201/04_box.txt)
![04_out](https://user-images.githubusercontent.com/28331455/121831230-2b441980-ccf1-11eb-9912-93c575011748.png)
[05_box.txt](https://github.com/cvlab-stonybrook/LearningToCountEverything/files/6645202/05_box.txt)
![05_out](https://user-images.githubusercontent.com/28331455/121831233-2c754680-ccf1-11eb-8e0b-508b6aa0d638.png)
[06_box.txt](https://github.com/cvlab-stonybrook/LearningToCountEverything/files/6645203/06_box.txt)
![06_out](https://user-images.githubusercontent.com/28331455/121831236-2d0ddd00-ccf1-11eb-994b-c4d46557a355.png)
[debug.log](https://github.com/cvlab-stonybrook/LearningToCountEverything/files/6645204/debug.log)

**Comment by Viresh-R** [comment]
Hey,
Before testing, you should resize the images so that the height of the test image is 384, and the width should be adjusted so as to preserve the aspect ratio of the image. FamNet was trained on images with height resized to 384, so it may perform poorly on larger images.

**Comment by leminhbk** [comment]
Thanks for your promptly response.
I tried to reduce the size of input images as you mentions. The results are slightly improved, but far from satisfactory.
It seems that the input images has high depth so the side of those fruits are so difference. Do I need to change the image to be more homogeneous or make change to other things?

![04s_out](https://user-images.githubusercontent.com/28331455/122019215-c3272d80-cded-11eb-9320-6f5ff508d412.png)
![03s_out](https://user-images.githubusercontent.com/28331455/122019223-c4f0f100-cded-11eb-9009-66466aba3280.png)
![02s_out](https://user-images.githubusercontent.com/28331455/122019225-c5898780-cded-11eb-8292-27ce38fc0e3a.png)
![01s_out](https://user-images.githubusercontent.com/28331455/122019227-c6221e00-cded-11eb-939a-a542a039f7fb.png)
![06s_out](https://user-images.githubusercontent.com/28331455/122019228-c6bab480-cded-11eb-8323-e49715e8cbf2.png)
![05s_out](https://user-images.githubusercontent.com/28331455/122019234-c7534b00-cded-11eb-8fab-29a31ea3c5e8.png)

**Comment by Viresh-R** [comment]
If the distribution of images on which you're testing our model are very different from the ones in our dataset, performance of FamNet is expected to get worse. That being said, here are few things you can try to improve the performance:
1. in case of large scale variation, try selecting exemplars from different scales (i.e. pick a large object, medium object and small object).
2. like you suggested, placing the camera so as to reduce the scale variation of the object should help.
3. if there's a specific category of objects on which you want to get accurate predictions, it would help if you can label some images from those categories  and finetune our model on those labeled images.

**Comment by iamfaith** [comment]
Hello, thanks for your great work. I try to fine-tune your model on my own dataset. How could I generate the density map? It seems that it is not open source yet. Thanks for your work. Any response will be appreciated a lot.

**Comment by Viresh-R** [comment]
Hey, 
If your dataset has dot annotations, you can use any of the publicly available crowd counting density map generation code. For ex, https://github.com/leeyeehoo/CSRNet-pytorch/blob/master/make_dataset.ipynb.
